docs/content/sips/008-logging.md

```diff
@@ -0,0 +1,104 @@
+title = "SIP 008 - Runtime Logging"
+template = "main"
+date = "2022-12-09T11:36:36Z"
+---
+
+Summary: Unified runtime logging
+
+Owner: mail@etehtsea.me
+
+Created: December 09, 2022
+
+## Background
+
+## Current experience
+
+Currently, by default spin prints into `stdout` only the basic info:
+
+```
+$ spin up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello: http://127.0.0.1:3000/hello
+    A simple component that returns hello.
+```
+
+Application log is implicitly written into `~/.spin/<spin_app>/logs/<component_id>_std{err|out}.log`.
+The log for the spin itself has log level `ERROR` and outputs into `stderr`.
+
+To enable a full logging experience, a developer needs to provide `--follow-all` to duplicate app log into `stderr` and also `RUST_LOG=spin=<log_level>` to see the spin log in the `stderr`.
+
+
+## Proposal
+
+- Set default log level to `INFO`.
+- Set default log destination to `stdout/stderr`.
+- Provide a way for spin apps for logging (besides using `println!`/`eprintln!`).
+- Log `WARN`and `ERROR` into `stderr` and `INFO`, `DEBUG`, `TRACE` into `stdout`.
```

Comment:
It should remain sequential if you redirect both somewhere (`spin up --follow-all &> out.log`).
But, currently, there is not way to do the opposite - for example, if I want to capture only warnings and errors on prod system. In this case I would redirect `stderr` somewhere: `spin up --follow-all 2&> err.log # or some fancy log collector instead of err.log`

But for now all output incl. `println!` from the guest modules (spin apps) is pointed to `stderr`.